### PR TITLE
chore: include suite, platform and tags in test reports name

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It adds a prefix to the test report names so that it's easier to look them up in the Jenkins UI. In the past, the report name was created calling `uname` in the remote VM, now it uses the metadata coming from the platform descriptors

## Why is it important?
Consistency about file names, and adding discoverability when reading reports in the Jenkins UI

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Related issues

- Completes #2245

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -528,6 +528,7 @@ def generateFunctionalTestStep(Map args = [:]){
   def envContext = []
   envContext.add("PROVIDER=${provider}")
   envContext.add("SUITE=${suite}")
+  envContext.add("REPORT_PREFIX=${suite}_${platform}_${tags}")
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:531]: Same as the parallel stage name

File: .ci/scripts/functional-test.sh [hunk 2]
```diff
@@ -25,13 +25,14 @@ STACK_VERSION=${3:-"${BASE_VERSION}"}
 BEAT_VERSION=${4:-"${BASE_VERSION}"}
 ELASTIC_AGENT_VERSION=${5:-"${BASE_VERSION}"}
 GOARCH=${GOARCH:-"amd64"}
+REPORT_PREFIX="${SUITE}_${GOARCH}_${TAGS}"
```
Comment (v1v) [inline .ci/scripts/functional-test.sh:28]: What do you think to allow overriding the value only if REPORT_PREFIX is not set? IIUC, 

https://github.com/elastic/e2e-testing/pull/2273/files#diff-ef8b2f5e4d21d81c0eba4754dea3a3dfc2c67d26284f195165b650cf4aff7beaR564 uses `.ci/scripts/functional-test.sh` with only one parameter, so `REPORT_PREFIX="${SUITE}_${GOARCH}_${TAGS}"` = `REPORT_PREFIX=${tags}_ amd64_" but `REPORT_PREFIX` is already in place as it's created on the fly in the jenkinsfile

Comment (mdelapenya) [reply .ci/scripts/functional-test.sh:28]: That could be fine too. My only concern is that the script that is run on CI is generated on the fly in the Ansible code, and this is different to the committed script. I'd try to align them as much as possible

**Review by kuisathaverat — APPROVED** [review]

File: .ci/scripts/functional-test.sh (hunk 2 above)
Comment (mdelapenya) [inline .ci/scripts/functional-test.sh:28]:
```suggestion
REPORT_PREFIX=${REPORT_PREFIX:-"${SUITE}_${GOARCH}_${TAGS}"}
```

Comment (mdelapenya) [reply .ci/scripts/functional-test.sh:28]: @v1v this is what you mean? I can enclose everything in quotes too, but I'm following same approach that in the above variables

**Pull request merged**